src/mono/mono/mini/monovm.c

```diff
@@ -117,13 +118,13 @@ mono_core_preload_hook (MonoAssemblyLoadContext *alc, MonoAssemblyName *aname, c
 	basename = g_strconcat (aname->name, ".dll", (const char*)NULL); /* TODO: make sure CoreCLR never needs to load .exe files */
 
 	for (int i = 0; i < a->assembly_count; ++i) {
-		if (!strcmp (basename, a->basenames[i])) {
+		if (!strcmp (basename, a->basenames [i])) {
```

Comment:
Strongly suggest to keep lengths around to speed up string compares..

Reply:
Agreed, will change.